packages/repository/src/relations/has-many/has-many-through.repository.ts

```diff
@@ -0,0 +1,59 @@
+import {Entity, DataObject, Options, Filter, Where, Count} from '../..';
+
+/**
+ * CRUD operations for a target repository of a HasManyThrough relation
+ */
+export interface HasManyThroughRepository<
+  Target extends Entity,
+  Through extends Entity
+> {
+  /**
+   * Create a target model instance
+   * @param targetModelData - The target model data
+   * @param throughModelData - The through model data
+   * @param options - Options for the operation
+   * @param throughOptions - Options passed to create through
+   * @returns A promise which resolves to the newly created target model instance
+   */
+  create(
+    targetModelData: DataObject<Target>,
```

Comment:
How to distingush between an existing or non-existing target entity? I think we need more operations in a `HasManyThroughRepository`. In `lb3` there was a link/unlink to create/delete a relation with an existing target entity.

Reply:
Thank you for the review. Would you like to see a method definition for “exists” as well as “link” and “unlink” (for example, https://apidocs.strongloop.com/loopback/#persistedmodel-exists)?

Reply:
@derdeka I added "add", "remove" and "exists" based on https://loopback.io/doc/en/lb3/HasManyThrough-relations.html and https://github.com/strongloop/loopback-datasource-juggler/blob/2f867576b66fe535aa1f82bceaaa680ecd4eec6c/lib/relation-definition.js.

Reply:
Example implementations of "add" and "remove" using the interface definition:

```
async add(
    targetModel: TargetEntity,
    throughModelData: DataObject<ThroughEntity> = {},
    throughOptions?: Options
  ): Promise<TargetEntity> {
    const throughRepository = await this.getThroughRepository();
    const throughConstraint = this.getThroughConstraint(targetModel);

    await throughRepository.create(
      constrainDataObject(
        throughModelData,
        throughConstraint as DataObject<ThroughEntity>
      ),
      throughOptions
    );

    return targetModel;
  }

  async remove(
    targetModel: TargetEntity,
    options?: Options,  // Note: maybe we don't need both options parameters?
    throughOptions?: Options
  ): Promise<void> {
    const throughRepository = await this.getThroughRepository();
    const throughConstraint = this.getThroughConstraint(targetModel);

    await throughRepository.deleteAll(
      constrainWhere(undefined, throughConstraint),
      throughOptions
    );
  }
```

Reply:
Sure, I added findById and destroyById and renamed the add/remove methods to link/unlink.